src/laminar.cpp

```diff
@@ -102,6 +102,8 @@ Laminar::Laminar(Server &server, Settings settings) :
              "PRIMARY KEY (name, number))");
     db->exec("CREATE INDEX IF NOT EXISTS idx_completion_time ON builds("
              "completedAt DESC)");
+    db->exec("CREATE INDEX IF NOT EXISTS idx_name_number ON builds("
+             "name, number DESC) WHERE result IS NOT NULL");
```

Comment:
Primary key index does not help, probably because it is missing `DESC`.
I can add `DESC` to the index, but that will require writing some migration to replace index for existing users.

Without my index the new queries take about the same time as the original query (between 2.5 and 5 seconds).
With index they take between 0.23 and 0.55 seconds.

Reply:
Then I think I would prefer having the migration. Otherwise we force everyone to have an expensive redundant index.

Reply:
Actually, the primary key also doesn't have `WHERE result IS NOT NULL` part. If the index does not have it, the query takes several seconds again (even if query plan is the same).

Reply:
Oh that's unfortunate. It got me thinking about why I check for NOT NULL anyway; I suspect it's a relic to maintain behaviour from before I stored incompleted runs in the database. I can't think of a good reason to prevent currently running builds from appearing on the jobs page, so maybe this is an opportunity to drop that check.

Reply:
Added the migration code.

I ran it on our database, but please also test it yourself before merging.